.github/workflows/test.yml

```diff
@@ -2,9 +2,16 @@ name: Run Unit Tests
 
 on: [pull_request]
 
+permissions:
+  contents: read
+
 jobs:
   test:
     runs-on: ubuntu-latest
+    permissions:
```

Comment:
Add comments on what each permission is allowing for better clarity.

Reply:
https://github.com/Azure/aks-async/pull/48/files#r1964254761